relayer/relayer/src/trader/mod.rs

```diff
@@ -0,0 +1,302 @@
+                            continue;
+                        }
+
+                        // check azero Eth balance
+                        let azero_eth_balance = match azero_ether
+                            .balance_of(azero_signed_connection.as_connection(), whoami_azero.clone())
+                            .await {
+                                Ok(balance) => balance,
+                                Err(why) => {
+                                    warn!("Error when querying for Azero ETH balance: {why:?}.");
+                                    continue;
+                                },
+                            };
+
+                        if azero_eth_balance > 0 {
+                            info!("Requesting a cross chain transfer of {azero_eth_balance} units of Azero ETH [{azero_ether_address}] to {whoami_eth}.");
+
+                            let mut receiver: [u8; 32] = [0; 32];
+                            receiver.copy_from_slice(&left_pad(eth_signed_connection.address().0.to_vec(), 32));
+
+                            // set allowance
+                            if let Err (why) = azero_ether.approve (&azero_signed_connection, most_azero.address.clone (), azero_eth_balance).await {
+                                warn!("Approve tx failed: {why:?}.");
+                                continue;
+                            }
+
+                            if let Err(why) = most_azero
+                                .send_request(
+                                    &azero_signed_connection,
+                                    *azero_ether.address.clone().as_ref(),
+                                    azero_eth_balance,
+                                    receiver,
+                                    current_base_fee
+                                )
+                                .await
+                            {
+                                warn!("Could not send the cross-chain transfer request: {why:?}.");
+                                continue;
+                            }
+
+                        }
+
+                    } else {
+                        debug!("{whoami_azero} has A0 balance too low for bridging");
+                    }
+
+                    // check 0xwETH balance
+                    let wrapped_ether_balance = match wrapped_ether
+                        .balance_of(eth_signed_connection.address())
+                        .block(BlockNumber::Finalized)
+                        .await
+                    {
+                        Ok(balance) => balance,
+                        Err(why) => {
+                            warn!("Query for WETH balance failed : {why:?}.");
+                            continue;
+                        }
+                    };
+
+                    debug!("{whoami_eth} has a balance of: {wrapped_ether_balance:?} WETH9.");
+
+                    // withdraw 0xwETH -> ETH
+                    if !wrapped_ether_balance.is_zero() {
```

Comment:
Since txns on Ethereum are quite expensive - maybe it's worth setting the surplus to something higher than non-zero?